### ISSUE TITLE
ERROR in Maximum call stack size exceeded

### ISSUE THREAD

**Issue by LucasBrazi06** [angular/angular-cli#6133]
### Bug Report or Feature Request (mark with an `x`)
```
- [ X ] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
@angular/cli: 1.0.1
node: 6.9.4
os: win32 x64
@angular/animations: 4.0.1
@angular/common: 4.0.1
@angular/compiler: 4.0.1
@angular/core: 4.0.1
@angular/forms: 4.0.1
@angular/http: 4.0.1
@angular/platform-browser: 4.0.1
@angular/platform-browser-dynamic: 4.0.1
@angular/router: 4.0.1
@angular/cli: 1.0.1
@angular/compiler-cli: 4.0.1
@angular/platform-server: 4.0.1

### Repro steps.
https://github.com/LucasBrazi06/ev-dashboard
(private: give me the person who will investigate, I'll add him in the contributor list)
This project has been migrated recently to angular-cli.

### The log given by the failure.
First time I compile I get this error message.

```
Hash: d018168a09703ca5d3fb
Time: 33317ms
chunk    {0} main.bundle.js, main.bundle.js.map (main) 204 kB {4} [initial] [rendered]
chunk    {1} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 167 kB {5} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 691 kB {5} [initial] [rendered]
chunk    {3} scripts.bundle.js, scripts.bundle.js.map (scripts) 491 kB {5} [initial] [rendered]
chunk    {4} vendor.bundle.js, vendor.bundle.js.map (vendor) 6.12 MB [initial] [rendered]
chunk    {5} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in Maximum call stack size exceeded
webpack: Failed to compile.

```

Next I just save a file which triggers the compilation again and it ends up successfully this time.

```
webpack: Compiling...
Hash: cc5121f438b5556e29ed
Time: 14094ms
chunk    {0} 0.chunk.js, 0.chunk.js.map 93.5 kB {1} {2} {3} [rendered]
chunk    {1} 1.chunk.js, 1.chunk.js.map 30.3 kB {0} {2} {3} [rendered]
chunk    {2} 2.chunk.js, 2.chunk.js.map 40.8 kB {0} {1} {3} [rendered]
chunk    {3} main.bundle.js, main.bundle.js.map (main) 204 kB {7} [initial] [rendered]
chunk    {4} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 167 kB {8} [initial] [rendered]
chunk    {5} styles.bundle.js, styles.bundle.js.map (styles) 691 kB {8} [initial] [rendered]
chunk    {6} scripts.bundle.js, scripts.bundle.js.map (scripts) 491 kB {8} [initial] [rendered]
chunk    {7} vendor.bundle.js, vendor.bundle.js.map (vendor) 6.12 MB [initial] [rendered]
chunk    {8} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]
webpack: Compiled successfully.
```

### Desired functionality.
Fix of the issue (surely in my project) and if something goes wrong during compil time, it would be nice to have more information to fix the issue.

**Comment by rudzikdawid** [comment]
Are you using lazy load with nested <router-outlet> ?

**Comment by LucasBrazi06** [comment]
Not in some files, so I assume that I must change this :-(

**Comment by LucasBrazi06** [comment]
Hi,
I refactored the project to change two components I used everywhere (toaster + confirm dialog) into services and inject them.
Now it works, it compiles correctly without this error message.
Nevetheless, I had to remove all the components and check part by part in order to identify the guilty ones.
It would have been great to have a tip in the error message about where to look at :-).
Thanks,
Serge.
